march_imu_manager/src/Mtw.cpp

```diff
@@ -0,0 +1,60 @@
+// Copyright 2019 Project March
+#include "march_imu_manager/Mtw.h"
+
+Mtw::Mtw(XsDevice* device, size_t maxBufferSize) :
+    m_device(device),
+    m_maxBufferSize(maxBufferSize)
+{
+    this->m_device->addCallbackHandler(this);
+    configure();
+}
+
+void Mtw::configure()
+{
+    XsOutputMode outputMode = XOM_Calibrated & XOM_Orientation;
+    XsOutputSettings outputSettings =
+        XOS_Timestamp_PacketCounter &
+        XOS_OrientationMode_Matrix &
+        XOS_CalibratedMode_AccGyrOnly;
+
+    this->m_device->setOutputMode(outputMode);
+    this->m_device->setOutputSettings(outputSettings);
+}
+
+bool Mtw::dataAvailable()
+{
+    std::unique_lock<std::mutex> lck(this->m_mutex);
+    return !this->m_packetBuffer.empty();
+}
+
+const XsDataPacket* Mtw::getOldestPacket()
+{
+    std::unique_lock<std::mutex> lck(this->m_mutex);
+    XsDataPacket const * packet = &this->m_packetBuffer.front();
+    return packet;
+}
+
+void Mtw::deleteOldestPacket()
```

Comment:
Wil je hier de packet niet ook gelijk returnen?

Reply:
It is not possible to return the package and delete it, otherwise the package will be deleted from the queue and therefore there exists no valid reference to it. You could do this by `std::move` the packet.